src/util/redisConnector.js

```diff
@@ -96,17 +110,32 @@ const RedisDB = {
   /**
    * Used to set value in redis depending on the key and the value type
    * @param {*} key key for which value needs to be stored
-   * @param {*} value value to be stored in redis
-   * @param {*} isValJson set to false if value is not a json object
+   * @param {*} value to be stored in redis send it in array format [field1, value1, field2, value2]
+   *                  if Value is an object
+   * @param {*} expiryTimeInSec expiry time of data in redis by default 1 hr
+   * @param {*} isValJson set to false if value is not a json object 
+   * 
    */
-
-  async setVal(key, value, isValJson = true) {
-    const dataExpiry = 60 * 60; // 1 hour
+  async setVal(key, value, expiryTimeInSec = 60 * 60) {
     try {
       await this.checkAndConnectConnection(); // check if redis is connected and if not, connect
-      const valueToStore = isValJson ? JSON.stringify(value) : value;
-      const bytes = Buffer.byteLength(valueToStore, 'utf-8');
-      await this.client.setex(key, dataExpiry, valueToStore);
+      let bytes;
+      if (typeof value === "object") {
+        const valueToStore = value.map(element => {
+          if (typeof element === "object") {
+            return JSON.stringify(element);
+          }
+          return element;
+        })
+        bytes = Buffer.byteLength(JSON.stringify(value), "utf-8");
+        await this.client.multi()
+          .hmset(key, ...valueToStore)
+          .expire(key, expiryTimeInSec)
+          .exec();
+      } else {
+        bytes = Buffer.byteLength(value, "utf-8");
```

Comment:
this is not required, aws provides this already

Reply:
okay

Reply:
done